dotnet-test-samples/apigw-lambda-ddb/src/ServerlessTestApi.Infrastructure/Startup.cs

```diff
@@ -17,27 +17,40 @@ public static ServiceProvider ServiceProvider
         {
             if (_serviceProvider == null)
             {
-                InitializeServiceProvider();
+                _serviceProvider = InitializeServiceProvider();
             }
 
             return _serviceProvider;
         }
         private set => _serviceProvider = value;
     }
 
-    private static void InitializeServiceProvider()
+    private static ServiceProvider InitializeServiceProvider()
     {
         var services = new ServiceCollection();
         
         var logger = new LoggerConfiguration()
             .WriteTo.Console(new RenderedCompactJsonFormatter())
             .CreateLogger();
         
-        services.AddLogging();
+        services.AddLogging(builder =>
+        {
+            builder.AddSerilog(logger);
+        });
+
+        if (Environment.GetEnvironmentVariable("USE_MOCKS") == "true")
```

Comment:
This is interesting - I wonder if we could iterate on this to remove the reference to mocks/Moq from the main application code. Using layers/extensions sounds like an obvious choice, but ideally we'd want to avoid pointing the main application at a random folder on the filesystem to load dependencies. Hooking events during the assembly load process might be helpful here.

Just food for thought - I think this is perfectly fine for a v1 implementation.

Reply:
Yeah, I absolutely want to iterate on it. I was going through the repo with a customer and they were asking how they could use the Visual Studio extension to debug locally. You can set that env var in the Visual Studio debug profile, giving a semi-function local dev experience.